packages/engine/src/datastore/table/pool/message.rs

```diff
@@ -75,19 +55,42 @@ impl MessagePool {
     }
 }
 
-// TODO: (clarity) Replace with MessageReader<'a> or rename ?
-pub struct MessagePoolRead<'a> {
-    batches: Vec<RwLockReadGuard<'a, MessageBatch>>,
+impl BatchPool<MessageBatch> for MessagePool {
+    fn read_proxy(&self) -> Result<PoolReadProxy<MessageBatch>> {
+        self.batches.iter().map(BatchReadProxy::new).collect()
+    }
+
+    fn partial_read_proxy(&self, indices: &[usize]) -> Result<PoolReadProxy<MessageBatch>> {
+        self.batches
```

Comment:
This seems slower than doing direct indexing. Was it just to satisfy borrow checking?

Reply:
This comes from the previous defaulted method on `BatchPool`, I haven't changed the logic. But you are probably right, that direct indexing is faster. We should do this in a follow up PR

Reply:
Can you make an asana task in the backlog if we don't want to driveby in this or the base PR